src/factory-stubs/faker.pyi

```diff
@@ -16,9 +16,8 @@ class Faker(Generic[T, V], declarations.BaseDeclaration[T, V]):
     @classmethod
     def override_default_locale(cls, locale: str) -> ContextManager[None]: ...
     @classmethod
-    @classmethod
     def _get_faker(cls, locale: str | None = ...) -> faker.Faker: ...
     @classmethod
     def add_provider(
-        cls, provider: faker.providers.BaseProvider, locale: str | None = ...
+        cls, provider: type[faker.providers.BaseProvider], locale: str | None = ...
```

Comment:
actually, it seems that it can be an instance or a subclass of BaseProvider.
```suggestion
        cls, provider: type[faker.providers.BaseProvider] | faker.providers.BaseProvider, locale: str | None = ...
```